### PR TITLE
Dynamic -OnChange- task rendering.

### PR DESCRIPTION
**Pull request by SarAhmed**
- Created Task interface.
- Create Item interface.
- Rendered tasks in the timeline
   - Add
   - Remove
   - Update

image

### CODE REVIEW

**Review by iguru-google — CHANGES_REQUESTED** [review]
Consider moving the demo in a separate path, so you only keep the component itself in src. This is not necessary now.

**Review by MW1245 — CHANGES_REQUESTED** [review]
Please include a short comment for each exported item.

**Review by MW1245 — APPROVED** [review]

**Review by iguru-google — APPROVED** [review]

File: timeline-dep-graph/src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,79 @@
...
+              dependants: [],
+              startTime: new Date('2020-09-28'),
+              finishTime: new Date('2020-09-29'),
+            },
+            ];
+          this.tasksDemo = tasksDemo4;
+          this.cdRef.detectChanges();
+        }, 2000);
+
+      }, 2000);
```
[the remark this answers is not in the document]
Comment (SarAhmed) [reply timeline-dep-graph/src/app/app.component.ts:75]: Implemented in: #22

File: timeline-dep-graph/src/app/app.component.ts [hunk 2]
```diff
@@ -1,10 +1,79 @@
...
+    id: '1',
+    name: 'Task 1',
+    dependants: [],
+    startTime: new Date('2020-09-28'),
+    finishTime: new Date('2020-09-29'),
+  }];
+
+  ngAfterViewInit(): void {
+    setTimeout(() => {
+      const tasksDemo2 = [ // Test Add
```
[the remark this answers is not in the document]
Comment (SarAhmed) [reply timeline-dep-graph/src/app/app.component.ts:24]: Implemented in: #22

File: timeline-dep-graph/src/app/app.component.ts [hunk 3]
```diff
@@ -1,10 +1,79 @@
...
+        },
+        {
+        id: '2',
+        name: 'Task 2',
+        dependants: [],
+        startTime: new Date('2020-09-28'),
+        finishTime: new Date('2020-09-30'),
+      }];
+      this.tasksDemo = tasksDemo2;
+      this.cdRef.detectChanges();
```
[the remark this answers is not in the document]
Comment (SarAhmed) [reply timeline-dep-graph/src/app/app.component.ts:40]: Implemented in: #22

File: timeline-dep-graph/src/app/app.component.ts [hunk 4]
```diff
@@ -1,10 +1,79 @@
...
+
+  tasksDemo: Task[] = [{ // Test Add
+    id: '1',
+    name: 'Task 1',
+    dependants: [],
+    startTime: new Date('2020-09-28'),
+    finishTime: new Date('2020-09-29'),
+  }];
+
+  ngAfterViewInit(): void {
```
[the remark this answers is not in the document]
Comment (SarAhmed) [reply timeline-dep-graph/src/app/app.component.ts:22]: Implemented in: #22

**Pull request merged**